osu.Game/Screens/Menu/Intro.cs

```diff
@@ -42,20 +43,24 @@ public class Intro : OsuScreen
         private Track track;
 
         [BackgroundDependencyLoader]
-        private void load(AudioManager audio, OsuConfigManager config, BeatmapManager beatmaps, Framework.Game game)
+        private void load(AudioManager audio, OsuConfigManager config, BeatmapManager beatmaps, Framework.Game game, OsuGame osuGame)
         {
             menuVoice = config.GetBindable<bool>(OsuSetting.MenuVoice);
             menuMusic = config.GetBindable<bool>(OsuSetting.MenuMusic);
 
             BeatmapSetInfo setInfo = null;
 
+            var currentRulesetId = osuGame.Ruleset.Value.ID;
+
             if (!menuMusic)
             {
-                var sets = beatmaps.GetAllUsableBeatmapSets();
+                var sets = beatmaps.GetAllUsableBeatmapSets().FindAll(b => b.Beatmaps.Exists(bm => bm.Ruleset.ID == 0 || bm.Ruleset.ID == currentRulesetId));

```

Comment:
I'm confused. I thought it's not respecting the current ruleset so why is it looking for "current or standard"?

Reply:
This shouldn't be required to be honest. The selection should just change when entering song select rather than being restricted at the `Intro`/`MainMenu`. There's many other flows which could cause this same problem to occur than this one.

Reply:
Song select already makes a correct selection, I guess then it should also update WorkingBeatmap if the current one is wrong ruleset?

Reply:
If you are able to enter song select and have a beatmaps with the non-current ruleset selected, this is definitely a bug. Should be fixed at song select and a test should be added to `TestCaseSongSelect` testing this specific case.